_includes/manuals/3.12/1.2_release_notes.md

```diff
@@ -4,16 +4,54 @@ This section will be updated prior to the next release.
 
 ### Headline features
 
+#### All Host index page improvements
+
+The experimental new All Hosts index page gets some major improvements with this release. To use the new page by default, in Settings, set Show new host overview page to Yes.
+The legacy All Hosts page and the new All Hosts page now have links to each other, so you can switch between them easily regardless of the setting value.
+Host bulk actions - You can now select one or more hosts and click the vertical ellipsis menu to perform the following actions:
+* Build management
+* Change host group
+* Delete
+Single-host actions - Additionally, you can click the vertical ellipsis in the rightmost table column (without selecting the host) and perform actions on a single host:
+* Edit
+* Clone
+* Delete
+Column selector - You can now click Manage Columns and customize columns displayed in the table. The column options are the same as those on the legacy All Hosts page. Any changes made here will also reflect on the legacy All Hosts page, and vice versa.
+
+For more details, see https://projects.theforeman.org/issues/37395, https://projects.theforeman.org/issues/37551, https://projects.theforeman.org/issues/37690, https://projects.theforeman.org/issues/37675, https://projects.theforeman.org/issues/37478, https://projects.theforeman.org/issues/37293.
+
+#### Puppet 8 support
+
+Foreman 3.12 now fully supports Puppet 8. While Foreman 3.11 unofficially supported Puppet 8, Puppet’s official puppetserver RPMs are known to break on Java. The installer now handles this for users. Users are encouraged to upgrade since Puppet 7 is expected to go end of life in February 2025.
+
+For more details, see https://projects.theforeman.org/issues/37686.
+
 ### Upgrade warnings
 
 ### Deprecations
 
+#### Running Foreman on {EL} 8 removal in Foreman 3.13
+
+Running on {EL} 8 is deprecated since Foreman 3.11.
+Foreman 3.13 will drop this support so users are encouraged to plan their {UpgradingDocURL}upgrading_EL_on_foreman_or_proxy_upgrading-connected[upgrade].
```

Comment:
`{UpgradingDocURL}` is also a macro that needs to be expanded here. You're also copying the asciidoc URL style that doesn't work with markdown.